### ISSUE TITLE
Enhancement: CalciteLink - add rel=”noopener noreferrer” for security

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-components#1647]
### Description
As noted by the security team, all hrefs in our apps should include `rel="noopener noreferrer"`.
Add this is a default attribute on the anchor element of CalciteLink.

Problem?

### Acceptance Criteria 
The attribute is on by default.
Should we have a prop to turn it off?

#### Which Component
CalciteLink

#### Example Use Case
Security in applications.

**Comment by julio8a** [comment]
This should only happen when there's a domain change? I remember @macandcheese implementing something like this in our old doc site links. We only applied to links going to external URLs.

**Comment by macandcheese** [comment]
I think this should be up to the user to add, currently it's not a defined prop but gets spread to the child link.

**Comment by paulcpederson** [comment]
`noopener` and `noreferrer` both only apply to cross domain links. Anything internal is sort of a moot point (ie. `esri.com/page-1/`links to `esri.com/page-2/`).

There are situations where you would probably want to keep the referrer. This would essentially allow other sites to see your referral traffic. If you put on `noreferrer` the site you link to will see the traffic reported as direct traffic. If you're trying to build a network to drive link authority, this would probably make that harder because your partners won't see the value you're providing them. This really only applies in SEO/Marketing scenarios.

`noopener` is the more important one, _especially_ if you're using `target="_blank"`. The page you linked to gets access to _your page_ via `window.opener`. This means they can do a whole bunch of nefarious things. 

Detecting `_target="blank"` and adding the rel attribute to the anchor would actually be a very nice security feature as it would prevent devs from leading their users to this kind of exploit. We could aways allow users to override that behavior by setting `rel` themselves like `<calcite-link rel="">`.

**Comment by dasa** [comment]
> `noopener` is the more important one, _especially_ if you're using `target="_blank"`.

It sounds like this may now be the browser's default: https://www.chromestatus.com/feature/6140064063029248

**Comment by benelan** [comment]
> Detecting `_target="blank"` and adding the rel attribute to the anchor would actually be a very nice security feature as it would prevent devs from leading their users to this kind of exploit. We could aways allow devs to override that behavior by setting `rel` themselves like `<calcite-link rel="">`.

@paulcpederson, Are there any action items for this issue? It looks like Chrome handles this by default as Dasa mentioned.

**Comment by paulcpederson** [comment]
I think it's perfectly valid to expect the user to set this. Doing nothing is an acceptable outcome IMO.

**Comment by benelan** [comment]
That seems like the consensus. I'll close this issue next week so @asangma if you want to chime in now's your chance.

**Comment by macandcheese** [comment]
I believe we would need to add explicit props here (and on button) like ionic does (https://github.com/ionic-team/ionic-framework/blob/25eb8cdf98fe455433ca6185e89d9e1223a6d3ae/core/src/components/button/button.tsx#L95) since the props are no longer spread, right?

**Comment by benelan** [comment]
Yeah good point. Seems like we have an action item so I won't close this one.
